docs/guidelines/PLUGIN_LIFECYCLE.md

```diff
@@ -0,0 +1,109 @@
+# Plugin Lifecytle
+
+Each plugin implements Init() and Close() method (see the [plugin_api..go](../../core/pluginapi.go)) 
+and optionally AfterInit(). Those methods are called sequncialy by (see the [agent_core.go](../../core/agent_core.go)).
+
+There are following rules for implementing the methods:
+## Init()
+* Initialize maps & channels here to avoid nil pointers later.
+* Process configs here (see the [Config Guidelins](CONFIG.md))
+* Propagate errors. Agent will stop because it is not properly initialized and calls Close() methods.
```

Comment:
If an error occurs, agent stops since it is not properly initialized and calls Close() methods.